backend/src/main/scala/cromwell/backend/WorkflowBackendActor.scala

```diff
@@ -0,0 +1,75 @@
+package cromwell.backend
+
+import akka.actor.{ActorRef, ActorLogging, Actor}
+import akka.event.LoggingReceive
+import cromwell.backend.WorkflowBackendActor._
+import cromwell.core.CallOutputs
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+import WorkflowBackendActor.ExecutionFailed
+
+object WorkflowBackendActor {
+
+  // Commands
+  sealed trait WorkflowBackendActorMessage
+
+  case class Execute(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+  case class Recover(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+  case class Abort(jobKey: BackendJobDescriptorKey) extends WorkflowBackendActorMessage
+
+  // Responses
+  sealed trait WorkflowBackendActorResponse
+
+  sealed trait ExecutionResponse extends WorkflowBackendActorResponse
+  case class ExecutionSucceeded(jobKey: BackendJobDescriptorKey, callOutputs: CallOutputs) extends ExecutionResponse
+  object ExecutionFailed {
+    def make(backendJobDescriptor: BackendJobDescriptor, throwable: Throwable) = ExecutionFailed(backendJobDescriptor.key, throwable)
```

Comment:
Oh this might be because the optimization I pasted before didn't quite work?  😄   You should be able to pass `ExecutionFailed.apply` instead of `ExecutionFailed.make` below, and then this object can go away.